packages/mdc-list/foundation.js

```diff
@@ -203,17 +204,31 @@ class MDCListFoundation extends MDCFoundation {
         this.adapter_.followHref(currentIndex);
       }
     }
+
+    if (isEnter || isSpace) {
```

Comment:
Would it be more straightforward to put this within the `else if` above, make that `else if` check only for `isEnter || isSpace`, then nest `if (this.isSingleSelectionList_)` within that? IMO it's a little hard to follow right now since there's two separate blocks with logic for enter/space, and potentially both of them can be executed.

Tangentially, I also can't remember why the preventDefault for single-selection is outside the isRootListItem check?